### PR TITLE
[stable25] Fix direct editing loading of txt files

### PR DESCRIPTION
**Pull request by juliushaertl**
Backport of #3476

### CODE REVIEW

File: src/services/SyncService.js [hunk 1]
```diff
@@ -95,12 +95,10 @@ class SyncService {
 			document: this.document,
 			session: this.session,
 		})
-		const content = connectionData.content
-			|| await this._fetchDocument()
 		this.emit('loaded', {
 			document: this.document,
 			session: this.session,
-			documentSource: '' + content,
+			documentSource: '' + connectionData.content,
```
Comment (marcelklehr) [inline src/services/SyncService.js:101]: Isn't this the same thing we removed above? I'm confused. :sweat_smile:

Comment (Raudius) [inline src/services/SyncService.js:101]: Minor nitpick, but could we just cast it?

```js
documentSource: String(content)
```

Comment (juliushaertl) [reply src/services/SyncService.js:101]: Yes, which is fine still just without the fetch fallback

Comment (juliushaertl) [reply src/services/SyncService.js:101]: The main problem was that an empty string (from a file without content) would always trigger the fetch request which is actually not necessary anymore as we always pass the content earlier.

**Review by Raudius — APPROVED** [review]

File: src/services/SyncService.js (hunk 1 above)
Comment (Raudius) [reply src/services/SyncService.js:101]: Nevermind, this is already a backport :see_no_evil: 

Approved

**Pull request merged**